### PR TITLE
functional and verbose parameters

### PR DESCRIPTION
**Pull request by traversc**
https://github.com/jhrcook/mustashe/issues/16

### CODE REVIEW

File: R/stash.R [hunk 1]
```diff
@@ -155,10 +162,15 @@ write_val <- function(var, val) {
...
+load_variable <- function(var, functional) {
   path <- stash_filename(var)$data_name
   val <- qs::qread(path)
-  assign_value(var, val)
+  if(functional) {
+    val
+  } else {
+    assign_value(var, val)
+    NULL
+  }
```
Comment (jhrcook) [inline R/stash.R]: Can you add explicit return calls? I think that would make the code a bit clearer and could potentially prevent some unexpected behavior (especially if new features are added).

File: R/unstash.R [hunk 2]
```diff
@@ -19,13 +20,13 @@
 #' }
 #'
 #' @export unstash
-unstash <- function(var) {
+unstash <- function(var, verbose=TRUE) {
```
Comment (jhrcook) [inline R/unstash.R]: Can you put a space between the variables, values and equal sign? `verbose = TRUE`. Thanks!

File: R/stash.R [hunk 3]
```diff
@@ -48,26 +50,31 @@ stash <- function(var, code, depends_on = NULL) {
...
 }
 
 # Make a new stash from a variable, code, and hash table.
-new_stash <- function(var, code, hash_tbl) {
+new_stash <- function(var, code, hash_tbl, functional) {
   val <- evaluate_code(code)
-  assign_value(var, val)
   write_hash_table(var, hash_tbl)
   write_val(var, val)
+  if(functional) {
```
Comment (jhrcook) [inline R/stash.R]: Can you add explicit return calls? I think that would make the code a bit clearer and could potentially prevent some unexpected behavior (especially if new features are added).

File: R/clear_stash.R [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 #'
 #' @return Returns \code{NULL} (invisibly).
 #'
 #' @examples
 #' clear_stash()
 #' @export clear_stash
-clear_stash <- function() {
-  message("Clearing stash.")
+clear_stash <- function(verbose=TRUE) {
+  if(verbose) message("Clearing stash.")
```
Comment (jhrcook) [inline R/clear_stash.R]: It's not the end of the world and you can leave it as is if you want, but I do prefer a space between the `if` and open ing parentheses and braces around the code block of an if statement: 

```r
if (verbose) { message("Clearing stash.") }
```

If you do make this change, can you update the other places, too? I'm thinking of adding a GH Action for styling the code in the future, so it really isn't a problem if you don't change this.

**Review by jhrcook — CHANGES_REQUESTED** [review]
I really like the additions and thank you for putting in a few tests for the new behavior. I left a few comments on the code, mainly minor styling that should be easy enough to quickly resolve. Thanks for the PR!

**Comment by traversc** [comment]
Changes made. Should be good to go now :)

**Review by jhrcook — APPROVED** [review]
Looks great!

**Pull request merged**